beacon_node/network/src/sync/block_lookups/tests.rs

```diff
@@ -29,15 +31,44 @@ type T = Witness<ManualSlotClock, CachingEth1Backend<E>, E, MemoryStore<E>, Memo
 struct TestRig {
     beacon_processor_rx: mpsc::Receiver<WorkEvent<E>>,
     network_rx: mpsc::UnboundedReceiver<NetworkMessage<E>>,
+    network_rx_queue: Vec<NetworkMessage<E>>,
     rng: XorShiftRng,
     harness: BeaconChainHarness<T>,
+    sync_manager: SyncManager<T>,
+    beacon_processor: Arc<NetworkBeaconProcessor<T>>,
 }
 
 const D: Duration = Duration::new(0, 0);
 
+/// This test utility enables integration testing of Lighthouse sync components.
+///
+/// It covers the following:
+/// 1. Sending `SyncMessage` to `SyncManager` to trigger `RangeSync`, `BackFillSync` and `BlockLookups` behaviours.
+/// 2. Making assertions on `WorkEvent`s received from sync
+/// 3. Making assertion on `NetworkMessage` received from sync (Outgoing RPC requests).
+///
+/// The test utility covers testing the interactions from and to `SyncManager`. In diagram form:
+///                      +-----------------+
+//                      | BeaconProcessor |
+//                      +---------+-------+
+//                             ^  |
+//                             |  |
+//                   WorkEvent |  | SyncMsg
+//                             |  | (Result)
+//                             |  v
+// +--------+            +-----+-----------+             +----------------+
+// | Router +----------->|  SyncManager    +------------>| NetworkService |
+// +--------+  SyncMsg   +-----------------+ NetworkMsg  +----------------+
+//           (RPC resp)  |  - RangeSync    |  (RPC req)
+//                       +-----------------+
+//                       |  - BackFillSync |
+//                       +-----------------+
+//                       |  - BlockLookups |
+//                       +-----------------+
```

Comment:
Since half is doc format and half is comment format this looks weird in my IDE, not sure if this is intentional: 

<img width="936" alt="Screenshot 2024-04-08 at 6 49 35 PM" src="https://github.com/sigp/lighthouse/assets/5160426/92df15e8-8382-4316-81db-13383916fcb5">

```suggestion
///                      | BeaconProcessor |
///                      +---------+-------+
///                             ^  |
///                             |  |
///                   WorkEvent |  | SyncMsg
///                             |  | (Result)
///                             |  v
/// +--------+            +-----+-----------+             +----------------+
/// | Router +----------->|  SyncManager    +------------>| NetworkService |
/// +--------+  SyncMsg   +-----------------+ NetworkMsg  +----------------+
///           (RPC resp)  |  - RangeSync    |  (RPC req)
///                       +-----------------+
///                       |  - BackFillSync |
///                       +-----------------+
///                       |  - BlockLookups |
///                       +-----------------+
```